composer.json

```diff
@@ -10,22 +10,35 @@
         {"name": "Roman Borschel", "email": "roman@code-factory.org"},
         {"name": "Benjamin Eberlei", "email": "kontakt@beberlei.de"},
         {"name": "Jonathan Wage", "email": "jonwage@gmail.com"},
-        {"name": "Johannes Schmitt", "email": "schmittjoh@gmail.com"}
+        {"name": "Johannes Schmitt", "email": "schmittjoh@gmail.com"},
+        {"name": "Fabio B. Silva", "email": "fabio.bat.silva@gmail.com"}
     ],
     "require": {
-        "php": ">=5.3.2",
-        "doctrine/lexer": "1.*"
+        "php":          ">=7.0.0",
+        "hoa/compiler": "3.*"
     },
     "require-dev": {
-        "doctrine/cache": "1.*",
-        "phpunit/phpunit": "4.*"
+        "doctrine/cache"                        : "1.*",
+        "phpunit/phpunit"                       : "5.*",
+        "instaclick/coding-standard"            : "~1.1",
+        "squizlabs/php_codesniffer"             : "~1.5",
+        "satooshi/php-coveralls"                : "~1.0",
+        "instaclick/object-calisthenics-sniffs" : "dev-master",
+        "instaclick/symfony2-coding-standard"   : "dev-remaster"
     },
     "autoload": {
-        "psr-0": { "Doctrine\\Common\\Annotations\\": "lib/" }
+        "psr-4": {
+            "Doctrine\\Annotations\\": "src/"
```

Comment:
@FabioBatSilva sed is not a friend to write code supporting both 1.x and 2.x at the same time. Your namespace change means you force the whole PHP community to upgrade to the new version at the same time (during the migration, there would be a netsplit in the community: migrated packages and not migrated packages being incompatible).
And the fact that the new version requires PHP 7 means that this split will live for a long time, as libraries can migrate to force using ``doctrine/annotations`` 2.x only when they themselves decide to drop support for PHP 5.x.

The key of an easy upgrade path is the possibility to easily write code compatible with both major versions of the library for most dependants.

But if you don't care about the fact that 359 open-source packages depend directly on your library (and so thousands of projects depend on it), please tell it explicitly and I will stop arguing about this point (but I can assure you that people won't be able to migrate to doctrine/annotations 2.x soon in that case for many of them)